.travis.yml

```diff
@@ -5,27 +5,45 @@ language: python
 python:
  - "3.5"
 
-addons:
-  apt:
-    packages:
-    - git
+jobs:
+  include:
+  - stage: tests
+    env: ENGINE
+    script:
+        - nosetests --with-doctest -xvs -a'!slow' openquake.engine
+        - mkdir ~/oqdata; nosetests --with-doctest -xvs -a'!slow' openquake.server
+        - bin/oq webui migrate
+        - cat /tmp/webui*
+    after_success:
+        - pip install sphinx==1.3.6 && cd doc/sphinx && make html
+  - stage: tests
+    env: HAZARDLIB
+    script:
+        - nosetests --with-doctest -xvs -a'!slow' openquake.hazardlib
+  - stage: tests
+    env: RISKLIB
+    script:
+        - nosetests --with-doctest -xvs -a'!slow' openquake.risklib
+  - stage: tests
+    env: COMMON
+    script:
+        - nosetests --with-doctest -xvs -a'!slow' openquake.commonlib
+        - nosetests --with-doctest -xvs -a'!slow' openquake.commands
+  - stage: tests
+    env: HMTK
+    script:
+        - nosetests --with-doctest -xvs -a'!slow' openquake.hmtk
+  - stage: demos
+    env: DEMOS
+    script: 
+        - bin/run-demos.sh
 
```

Comment:
There are too many stages. I would keep only three:

TESTS/HAZARDLIB:
   baselib, hazardlib, hmtk
TESTS/ENGINE:
   engine,server,calculators,risklib,commonlib,commands 
DEMOS:
     as it is